miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java

```diff
@@ -995,6 +1001,13 @@ private static DetailedLibraryDto asDetailedLibraryDto(DetailedLibrary from) {
     }
     if (from.getSample() != null) {
       dto.setIdentityConsentLevel(getIdentityConsentLevelString((DetailedSample) from.getSample()));
+      if (isDetailedSample(from.getSample())) {
```

Comment:
You could remove this `if` statement as there are many checks in place to ensure a DetailedLibrary isn't parented to a Plain Sample.